daemon/smartcard.c

```diff
@@ -177,6 +177,33 @@ smartcard_send_recv_block(const DaemonToToken *out)
 	return msg;
 }
 
+bool
+smartcard_cert_has_valid_format(unsigned char *cert_buf, size_t cert_buf_len)
+{
+	const char *begin_cert_str = "-----BEGIN CERTIFICATE-----\n";
+	const char *end_cert_str = "-----END CERTIFICATE-----\n";
+	size_t begin_cert_str_len = strlen(begin_cert_str);
+	size_t end_cert_str_len = strlen(end_cert_str);
+
+	if (cert_buf == NULL || cert_buf_len == 0) {
+		ERROR("Certificate not provided.");
```

Comment:
May an error message like "Given certificate is empty" would be clearer

Reply:
Fixed in the latest commit